xbmc/settings/Settings.cpp

```diff
@@ -1012,7 +1012,7 @@ void CSettings::InitializeISettingCallbacks()
   settingSet.clear();
   settingSet.insert(CSettings::SETTING_WEATHER_ADDON);
   settingSet.insert(CSettings::SETTING_WEATHER_ADDONSETTINGS);
-  GetSettingsManager()->RegisterCallback(&g_weatherManager, settingSet);
+  GetSettingsManager()->RegisterCallback(&CServiceBroker::GetWeatherManager(), settingSet);
```

Comment:
This is not correct. @Montellese this mechanism for registering callbacks is based on globals and has become an issue now. Here it registers a method that does not exist at this time.
We need a mechanism where services can register dynamically.

Reply:
I didn't think of it while refactored the weather manager. We are able to call `RegisterCallback` from `CWeatherManager`. There is no need to leave it here.

Reply:
Nice, I didn't know. Are you taking care of it?

Reply:
see https://github.com/xbmc/xbmc/pull/13326